src/sources/kubernetes_logs/mod.rs

```diff
@@ -618,6 +633,18 @@ fn prepare_label_selector(config: &Config) -> String {
     format!("{},{}", BUILT_IN, config.extra_label_selector)
 }
 
+// This function constructs the effective namespace label selector to use, based on
+// the specified configuration.
+fn prepare_namespace_label_selector(config: &Config) -> String {
```

Comment:
This _feels_ like we could update the `prepare_label_selector` to be used for either pod labels or namespace labels since they're so similar.

Reply:
I thought about just reusing the `prepare_label_selector`, but wasn't too sure about it.
Would you prefer the `let namespace_label_selector = prepare_label_selector(config);`?

Reply:
ah.. right, the `prepare_label_selector` would have to be changed to pick `config.extra_label_selector` or `config.extra_namespace_label_selector` depending on whether we're targeting Pods or Namespaces

Reply:
Yep - hadn't fully though what the new function would look like to do so properly, but it seemed like a good option instead of duplicating 👍

Reply:
I'll give this some thought and ask around the team for suggestions as well - of course, feel free to tackle it yourself if you have a good idea

Reply:
```rust
// This function constructs the effective label selector to use, based on
// the specified configuration.
fn prepare_label_selector(selector: String) -> String {
    const BUILT_IN: &str = "vector.dev/exclude!=true";

    if selector.is_empty() {
        return BUILT_IN.to_string();
    }

    format!("{},{}", BUILT_IN, selector)
}
```

And then updating the callsite to pass in `config.extra_blah...` rather than the entire `config` struct.

Does that seem reasonable @anapsix

Reply:
Thanks @spencergilbert. I've made an attempt to apply your suggestion.